llvm/test/tools/llvm-symbolizer/debug-info-line-info.yaml

```diff
@@ -0,0 +1,175 @@
+# Test llvm-symbolizer always uses line info from debug info if present.
+# RUN: yaml2obj %s -o %t
+# RUN: llvm-symbolizer --obj=%t 0x1 | FileCheck %s
+
+# CHECK:      foo(bool)
+# CHECK-NEXT: ??:0:0
+
+--- !ELF
+FileHeader:
+  Class:           ELFCLASS64
+  Data:            ELFDATA2LSB
+  Type:            ET_REL
+  Machine:         EM_X86_64
+  SectionHeaderStringTable: .strtab
+Sections:
+  - Name:            .text
+    Type:            SHT_PROGBITS
+    Flags:           [ SHF_ALLOC, SHF_EXECINSTR ]
+    AddressAlign:    0x10
+    Content:         50E80000000031C059C3
+  - Name:            .debug_abbrev
+    Type:            SHT_PROGBITS
+    AddressAlign:    0x1
+    Content:         01110025251305032572171017111B12067317000000
```

Comment:
Perhaps use an assembly test (reduced from `clang -S`, with post modification) which has better readability

Reply:
I found that I have to use yaml file as test case. If I use assembly test, when I use llvm-mc or clang to compile it to object file, the object file will have the current cwd as the directory name. Then this will make the llvm-symbolizer passes this test as well, because https://github.com/llvm/llvm-project/blob/af7c8c475ab97382a466bce8ad2fd091e0e09c1e/llvm/lib/DebugInfo/Symbolize/SymbolizableObjectFile.cpp#L323 condition will always be false when the full path file name is "<dir>/<invalid>".

Reply:
Then how did/does this come up in practice?

Reply:
Looked again, I think this change is unnecessary as this doesn't happen in practice. See detailed comment below.

Reply:
Bit of an awkward test to depend on implementation details to match the file name to the invalid value LLVM uses internally, etc... 

But I guess without it, this patch is untestable/has no intentionalyl observable behavior without #124846 ... *shrug*